dCommon/GeneralUtils.h

```diff
@@ -123,90 +125,117 @@ namespace GeneralUtils {
 
 	std::vector<std::string> GetSqlFileNamesFromFolder(const std::string& folder);
 
+	// Concept constraining to enum types
 	template <typename T>
-	T Parse(const char* value);
+	concept Enum = std::is_enum_v<T>;
 
-	template <>
-	inline bool Parse(const char* value) {
-		return std::stoi(value);
-	}
+	// Concept constraining to numeric types
+	template <typename T>
+	concept Numeric = std::integral<T> || Enum<T> || std::floating_point<T>;
 
-	template <>
-	inline int32_t Parse(const char* value) {
-		return std::stoi(value);
-	}
+	// Concept trickery to enable parsing underlying numeric types
+	template <Numeric T>
+	struct numeric_parse { using type = T; };
 
-	template <>
-	inline int64_t Parse(const char* value) {
-		return std::stoll(value);
-	}
+	// If an enum, present an alias to its underlying type for parsing
+	template <Numeric T> requires Enum<T>
+	struct numeric_parse<T> { using type = std::underlying_type_t<T>; };
 
-	template <>
-	inline float Parse(const char* value) {
-		return std::stof(value);
-	}
+	// If a boolean, present an alias to an integral type for parsing
+	template <Numeric T> requires std::same_as<T, bool>
+	struct numeric_parse<T> { using type = uint32_t; };
 
-	template <>
-	inline double Parse(const char* value) {
-		return std::stod(value);
-	}
+	// Shorthand type alias
+	template <Numeric T>
+	using numeric_parse_t = numeric_parse<T>::type;
 
-	template <>
-	inline uint16_t Parse(const char* value) {
-		return std::stoul(value);
-	}
+	/**
+	 * For numeric values: Parses a C-style char range (string) and returns an optional variable depending on the result
+	 * @param str The pointer to the start of the char range (string)
+	 * @param strEnd The pointer to the end of the char range (string), defaults to NULL
+	 * @returns An std::optional containing the desired value if it exists in the string
+	*/
+	template <Numeric T>
+	[[nodiscard]] std::optional<T> TryParse(const char* const str, const char* const strEnd = NULL) noexcept {
+		numeric_parse_t<T> result;
+		const bool isParsed = std::from_chars(str, strEnd, result).ec == std::errc{};
 
-	template <>
-	inline uint32_t Parse(const char* value) {
-		return std::stoul(value);
+		return isParsed ? static_cast<T>(result) : std::optional<T>{};
 	}
 
-	template <>
-	inline uint64_t Parse(const char* value) {
-		return std::stoull(value);
-	}
+#ifdef __APPLE__
```

Comment:
I would include `dPlatforms.h` and use the macros in there for the platform type instead.